nautobot_secrets_providers/providers/hashicorp.py

```diff
@@ -178,16 +191,28 @@ def get_value_for_secret(cls, secret, obj=None, **kwargs):
         """Return the value stored under the secret’s key in the secret’s path."""
         # Try to get parameters and error out early.
         parameters = secret.rendered_parameters(obj=obj)
+        try:
+            configuration = parameters.get("hashicorp_config", "default")
+            vault_settings = settings.PLUGINS_CONFIG["nautobot_secrets_providers"].get("hashicorp_vault", {})
+            if configuration and configuration != "default":
+                vault_settings = settings.PLUGINS_CONFIG["nautobot_secrets_providers"]["hashicorp_vault"][
+                    "configurations"
+                ].get(configuration, {})
+        except KeyError:
+            vault_settings = {}
+        secret_mount_point = vault_settings.get("default_mount_point", "secret")
+        secret_kv_version = vault_settings.get("kv_version", HashicorpKVVersionChoices.KV_VERSION_2)
+
         try:
             secret_path = parameters["path"]
             secret_key = parameters["key"]
-            secret_mount_point = parameters.get("mount_point", DEFAULT_MOUNT_POINT)
-            secret_kv_version = parameters.get("kv_version", DEFAULT_KV_VERSION)
+            secret_mount_point = parameters.get("mount_point", secret_mount_point) or secret_mount_point
+            secret_kv_version = parameters.get("kv_version", secret_kv_version) or secret_kv_version
```

Comment:
Should these defaults remain constants?

Reply:
They are not global constants anymore. Now they are configured per Vault, each HashiCorp Vault instance can have their own mount_point and kv_version